packages/react-form/src/hooks/field/reducer.ts

```diff
@@ -88,10 +90,33 @@ export function makeFieldReducer<Value>({
       }
 
       case 'updateError': {
-        return {
-          ...state,
-          error: action.payload,
-        };
+        if (Array.isArray(action.payload)) {
```

Comment:
You could combine the array and non-array cases here, with something like this:

```
const errors = Array.isArray(action.payload) ? action.payload : [action.payload];
```

Then from the non you can use `errors` instead of `action.payload`.

Reply:
Like @michaelmelanson said,

you can do : 
```ts
const payload = Array.isArray(action.payload) ? action.payload : [action.payload];
const [firstError] = payload;
if (shallowArrayComparison(action.payload, state.allErrors)) {
  return {
      ...state,
      error: firstError,
  };
} else {
  return {
    ...state,
    error: firstError,
    allErrors: action.payload,
  };
}
```